src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -495,7 +498,13 @@ internal void CheckConstraintLanguageVersionAndRuntimeSupportForConversion(Synta
             var builder = ArrayBuilder<BoundExpression>.GetInstance(elements.Length);
             foreach (var element in elements)
             {
-                builder.Add(convertArrayElement(element, elementType, diagnostics));
+                var result = element switch
+                {
+                    BoundCollectionLiteralSpreadElement spreadElement => throw ExceptionUtilities.UnexpectedValue(spreadElement),
```

Comment:
prototype comment?  why is this unexpected?

Reply:
If the array contains any spread elements, it is handled with a `List<T>` at line 489 above.

Reply:
Correct, we are not optimizing for that case yet.